scripts/randdm

```diff
@@ -0,0 +1,115 @@
+                   'Options: "band" or "block". If not specified, then '
+                   'draws from gaussian distribution to generate distance '
+                   'matrices.', default=None)
+@click.option('--dim', 'dimensions', type=int, multiple=True,
+              help='Dimension of the input matrix/matrices to generate')
+@click.option('--seed', 'seed', type=int,
+              help='Random number generator seed value.')
+@click.option('--sub', 'subsample_dims', type=str, multiple=True,
+              help='Generate subsampled distance matrix with given'
+                   'dimension (must be smaller than original dimension)'
+                   ' for each randomly generated distance matrix.'
+                   ' Can specify as int or percentage')
+@click.option('--overwrite', 'overwrite', type=bool, default=False,
+              help='Overwrite output directory if it already exists. Do not '
+                   'overwrite, by default.')
+@click.argument('output_dir', type=click.Path(), default=None)
+def generate(dimensions, output_dir, seed, subsample_dims, structure,
+             overwrite):
+    """
+    Generate random distance matrix
+    """
+
+    if seed is not None:
+        np.random.seed(seed)
+
+    if (not overwrite and os.path.exists(output_dir) and not click.confirm(
+            'The output directory %s exists. '
+            'Do you want to overwrite?' % output_dir)):
+        click.echo('Aborting.')
+        return
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    for dim in dimensions:
+        outpath = os.path.join(output_dir, 'randdm-{}.txt'.format(dim))
+
+        click.echo('Generating random distance matrix: %s' % outpath)
+
+        # Generate random distance matrix
+        if structure is not None:
+            click.echo('Generating synthetic OTU table with "{}"-like '
+                       'microbial distribution...'.format(structure))
+            if structure == 'band':
+                biom_table = band_table(dim, dim, seed=seed)[0]
+            elif structure == 'block':
+                biom_table = block_table(dim, dim, seed=seed)[0]
+            else:
+                raise ValueError('Invalid value for --structure parameter.')
+
+            otu_table = biom_table.matrix_data.todense()
+
+            click.echo('Generating distance matrix from OTU table...')
+            distance_matrix = beta_diversity('braycurtis', otu_table,
+                                             validate=False)
+
+        else:
+            click.echo('Generating random distance matrix '
+                       '(gaussian distribution)')
+            distance_matrix = randdm(dim, constructor=DistanceMatrix)
+
+        # Serialize distance matrix
+        distance_matrix.write(outpath)
+
+        # Subsampling
+        for subsample_dim in subsample_dims:
+            # Parse parameter values into integers
+            if '%' in subsample_dim:
```

Comment:
So in theory you can pass: 100%, 200%, -10001%, XX for this value ... perhaps worth adding a nicer validation step so it doesn't brake cause it can't transform to int; this applies to all parameters.